open_data_schema_map.file_cache.inc

```diff
@@ -31,19 +57,78 @@ function open_data_schema_map_file_cache_exits($api, $arguments = array()) {
 function open_data_schema_map_file_cache_endpoint($machine_name) {
   $api = open_data_schema_map_api_load($machine_name);
   if (isset($api)) {
-    drush_log('API present and loaded', 'success');
+    watchdog('open_data_schema_map', 'api present and loaded', WATCHDOG_INFO);
     $filename = open_data_schema_map_file_cache_name_helper($api);
     // This will only work for data.json v1.1 right now.
-    drush_log('Beginning processing for endpoint (this could take a while)', 'success');
+    watchdog('open_data_schema_map', 'Beginning processing for endpoint (this could take a while)', WATCHDOG_INFO);
     $render = open_data_schema_map_render_api($api);
     $result = $render['result'];
 
     // Load the correct output format, render results, and write to a file.
     if ($output_format = open_data_schema_map_output_format_load($api->outputformat)) {
       $response = $output_format['callback']($api, $result);
-      drush_log('Saving ' . $filename, 'success');
+      watchdog('open_data_schema_map', "Saving $filename", WATCHDOG_INFO);
       file_save_data($response, $filename, FILE_EXISTS_REPLACE);
-      drush_log($filename . ' saved', 'success');
+      watchdog('open_data_schema_map', "$filename saved", WATCHDOG_INFO);
     }
   }
 }
+
+/**
+ * API wrapper for open_data_schema_map_file_cache_endpoint.
+ */
+ function open_data_schema_map_file_cache_create($api) {
+    open_data_schema_map_file_cache_endpoint($api->machine_name);
+ }
+
+
+/**
+ * Delete a specific api endpoint file cache.
+ */
+function open_data_schema_map_file_cache_delete($api) {
+  $filename = open_data_schema_map_file_cache_name_helper($api);
+  if (file_exists($filename)) {
+    file_unmanaged_delete($filename);
+  }
+}
+
+/**
+ * Callback for the api endpoint file cache admin page.
+ */
+function open_data_schema_map_cache_manage(array $form, array &$form_state, $api = NULL) {
```

Comment:
Two things - I think this page callback should go in the .pages.inc file. Also, we need some help text on that page because it's totally unclear to a user what's going on here. I would add:

The Open Data Schema Map filecache writes an open data endpoint such as data.json to disk, so that it does not need to be generated upon every request. See the [documentation](http://docs.getdkan.com/en/latest/components/open-data-schema.html#the-odsm-file-cache) for instructions on automating this process. Use the controls on this page to manually generate or remove this API's filecache.

**Filecache status:** Not cached `||` 2 hours ago

[Buttons]

Reply:
I personally think `open_data_schema_map_cache_manage` should remain in file_cache.inc because it keeps all the file_cache functions in one place.